tower-grpc-interop/src/client.rs

```diff
@@ -278,33 +318,202 @@ impl Testcase {
-                    util::client_payload(1828),
-                    util::client_payload(45904),
-                ]);
-                core.run(
-                    client.streaming_input_call(Request::new(stream))
-                        .then(|result| {
-                            let mut assertions = vec![
-                                    test_assert!(
-                                        "call must be successful",
-                                        result.is_ok(),
-                                        format!("result={:?}", result)
-                                    )
-                            ];
-                            if let Ok(response) = result.map(|r| r.into_inner()) {
-                                assertions.push(test_assert!(
-                                "aggregated payload size must be 74922 bytes",
-                                response.aggregated_payload_size == 74922,
-                                format!("aggregated_payload_size={:?}",
-                                    response.aggregated_payload_size
-                                )));
+                let stream = stream::iter_ok(REQUEST_LENGTHS.iter()
+                    .map(|len| util::client_payload(*len as usize)));
+                core.run(client
+                    .streaming_input_call(Request::new(stream))
+                    .then(|result| {
+                        let mut assertions = vec![
+                                test_assert!(
+                                    "call must be successful",
+                                    result.is_ok(),
+                                    format!("result={:?}", result)
+                                )
+                        ];
+                        if let Ok(response) = result.map(|r| r.into_inner()) {
+                            assertions.push(test_assert!(
+                            "aggregated payload size must be 74922 bytes",
+                            response.aggregated_payload_size == 74922,
+                            format!("aggregated_payload_size={:?}",
+                                response.aggregated_payload_size
+                            )));
+                        }
+                        future::ok::<Vec<TestAssertion>, Box<Error>>(assertions)
+                    }))
+            },
+            Testcase::server_streaming => {
+                use pb::ResponseParameters;
+                let req = pb::StreamingOutputCallRequest {
+                    response_parameters: RESPONSE_LENGTHS.iter().map(|len| {
+                        ResponseParameters::with_size(*len)
+                    }).collect(),
+                    ..Default::default()
+                };
+                let req = Request::new(req);
+                core.run(client
+                    .streaming_output_call(req)
+                    .map_err(|tower_err: tower_grpc::Error<tower_h2::client::Error>| -> Box<Error> {
```

Comment:
I think this closure can be written as just `.map_err(Box::new)`, which is somewhat more concise.

Reply:
That does not work, the compiler doesn't understand that I want this closure's return type to be `Box<Error>` (rather than `Box<tower_grpc::Error<...>>`) then.

Reply:
I think this could be improved with #6. I think tower-grpc would be well served by only one error type; this is how the C++ gRPC library works.

Reply:
@per-gron 
> That does not work, the compiler doesn't understand that I want this closure's return type to be `Box<Error>` (rather than `Box<tower_grpc::Error<...>>`) then.

Ah, okay. In that case, I _think_ you could write `.map_err(Box::<Error>::new)`, which is still slightly more concise --- I haven't actually checked to make sure the type checker will accept that, but I think it should...